custom_messages/nodes/fiducial_tracker.py

```diff
@@ -0,0 +1,284 @@
+                linSpeed = 0
+                angSpeed = 0
+            # A fiducial was detected since last iteration of this loop
+            elif self.got_fid:
+                # Set the turning speed based on the angular error
+                # Add some damping based on the previous speed to smooth the motion 
+                angSpeed = angular_error * self.angular_rate - angSpeed / 2.0
+                # Make sure that the angular speed is within limits
+                if angSpeed < -self.max_angular_rate:
+                    angSpeed = -self.max_angular_rate
+                if angSpeed > self.max_angular_rate:
+                    angSpeed = self.max_angular_rate
+
+                # Set the forward speed based distance
+                linSpeed = forward_error * self.linear_rate
+                # Make sure that the angular speed is within limits
+                if linSpeed < -self.max_linear_rate:
+                    linSpeed = -self.max_linear_rate
+                if linSpeed > self.max_linear_rate:
+                    linSpeed = self.max_linear_rate
+
+            # Hysteresis, don't immediately stop if the fiducial is lost
+            elif not self.got_fid and times_since_last_fid < self.hysteresis_count:
+                # Decrease the speed (assuming linear decay is <1)
+                linSpeed *= self.linear_decay
+
+            # Try to refind fiducial by rotating
+            elif self.got_fid == False and times_since_last_fid < self.max_lost_count:
+                # Stop moving forward
+                linSpeed = 0
+                # Keep turning in the same direction
+                if angSpeed < 0:
+                    angSpeed = -self.lost_angular_rate
+                elif angSpeed > 0:
+                    angSpeed = self.lost_angular_rate
+                else:
+                    angSpeed = 0
+                print "Try keep rotating to refind fiducial: try# %d" % times_since_last_fid
+            else:
+                angSpeed = 0
+                linSpeed = 0
+
+            print "Speeds: linear %f angular %f" % (linSpeed, angSpeed)
+
+            # Create a Twist message from the velocities and publish it
+            # Avoid sending repeated zero speed commands, so teleop
+            # can work
+            zeroSpeed = (angSpeed == 0 and linSpeed == 0)
+            if not zeroSpeed:
+                self.suppressCmd = False
+            print "zero", zeroSpeed, self.suppressCmd
+            if not self.suppressCmd:
+                twist = Twist()
+                twist.angular.z = angSpeed
+                twist.linear.x = linSpeed
+                self.cmdPub.publish(twist)
+                if zeroSpeed:
+                    self.suppressCmd = True
+
+            # We already acted on the current fiducial
+            self.got_fid = False
+            rate.sleep()
+
+
+if __name__ == "__main__":
+    # Create an instance of our follow class
+    node = Follow()
+    # run it
+    node.run()
```

Comment:
This file looks like it has a lot of similarity with the other fiducial tracker.  Could we import that into this file to avoid the duplicated code?   Or could this version replace the original?

Reply:
The .pyc file should not be in the repo.

Reply:
This new follow.py completely removes the need to have what I had earlier called fiducial_tracker.   So there is not going to be a fiducial tracker, only an improvement you see here to our legacy fiducial follow.